### DISCUSSION TITLE
Zebrafish Behavior Data question

### DISCUSSION THREAD

**Discussion by ucscbrianlee** [NeurodataWithoutBorders/helpdesk#91] [Q&A]
Dear NWB community,

I'm looking at data coming off a ZebraScope and processed in ZebraLab software that tracks zebrafish behavior larvae in 96 well plates.

ZebraLab stores the raw data in a proprietary file format with a .RAW extension.   The software then produces a large number of .XLS files.  Each of these Excel files contains detailed data about the experiment, such as frame-by-frame information for each well in a multi-well plate setup.  These Excel files are used for further analysis and processing, allowing researchers to quantify the behavior of zebrafish by examining metrics.

Here is an example of a triplicate recording "190701" of one zebrafish mutant (many larvae) and the files related to the output:

```
dyrk1a_4ins_del5 folder contains three folders:  190701_0B  190701_0C  190701_0D

dyrk1a_4ins_del5/190701_0B:
190701_0B_off.raw  190701_0B_off.XLS  190701_0B_on.raw  190701_0B_on.XLS  190701_0B.XLS

dyrk1a_4ins_del5/190701_0C:
190701_0C_off.raw  190701_0C_off.XLS  190701_0C_on.raw  190701_0C_on.XLS  190701_0C.XLS

dyrk1a_4ins_del5/190701_0D:
190701_0D_off.raw  190701_0D_off.XLS  190701_0D_on.raw  190701_0D_on.XLS  190701_0D.XLS
```

This is Zebrafish behavioral data, but it does not have a clear component seen in NWB conversion:

That is to say there is no data like:
```
    data=[1, -1, 1, -1],
    timestamps=[15.0, 30.0, 60.0, 95.0],
```

Instead the .RAW is not a movie, where NWB requires ` VIDEO_FILE_EXTENSIONS = [".mp4", ".avi", ".wmv", ".mov", ".flv", ".mkv"]` and the excel files have lines like this:

location | animal | user | sn | an | datatype | start | end | startreason | endreason | frect | fredur | midct | middur | burct | burdur | zerct | zerdur | actinteg | stdate | sttime
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
w001 | A01 | ZEB8139-4-PC\ZEB8139-4 | 1 | 0 | quantauc | 0 | 60 | End of period | End of period | 2 | 59.9 | 2 | 0.2 | 0 | 0 | 2 | 59.7 | 76 | 1/7/19 | 15:38:52
w002 | A02 | ZEB8139-4-PC\ZEB8139-4 | 1 | 0 | quantauc | 0 | 60 | End of period | End of period | 0 | 60 | 0 | 0 | 0 | 0 | 0 | 60 | 0 | 1/7/19 | 15:38:52  

Also, beyond this the lab has a summary table with information like this:

| Mutant Line	| Behavioral Parameter | 	WT_Mean	| WT_SD	| Mutant_Mean	| Mutant_SD	| Beta |	Pvalue |
| --	| --	| --	| -- |	-- | -- | -- | -- |
| HET: chd8_del5	| OFF_POST	| *31.59122257*	| *27.38104973*	| 30.48859416	| 26.86102982	| 1.365310295 |	0.545167426 |
| --	| --	| --	| -- |	-- | -- | -- | -- |
| HOM: chd8_del5	 | OFF_POST	| *31.59122257*	| *27.38104973*	| 31.56431964 |	28.47617429	| 2.303537016	| 0.394848586 |

When I am looking at this data and the NWB format, I am not seeing the path of how I can convert this and the consideration of placing the files in Zenodo has come up as an alternative.

I have gone looking for NWB zebrafish behavior files and found this DANDISET: https://dandiarchive.org/dandiset/000350?search=zebrafish+behavior&pos=1

After some investigation I belive it goes back to this GitHub scripted by @CodyCBakerPhD for conversion: 
* https://github.com/catalystneuro/ahrens-lab-to-nwb
* https://github.com/catalystneuro/ahrens-lab-to-nwb/issues/3

This at first gives me hope that perhaps there is a way forward to collapse all the files for this experiment into NWB (the .raw, .xls, and the final analysis .csv) but I am still not sure it is possible? 

I would greatly welcome and appreciate any guidance.

**Comment by ucscbrianlee** [comment]
@bendichter  
Hi Ben,

There is some time pressure on this topic, could you assist in feedback/guidance.  While Zenodo was considered as an alternative for this Zebrafish Behavioral data it does not sound like that will be acceptable.   Conceptually it seems like some of this data should be convertible into a DANDI format.  My review of the tutorials/neuroconv examples isn't outlining the path forward (if even feasible?).   Also, on a conceptual level, when I look at the dandiset 000350 that apparently has Zebrafish Behavioral data, I am not fully understanding how to extract that data from the NWB file -just discussing how that dataset (NWB conversions) occurred and how to access the data within those NWB files would be of great interest and help.

Sincerely,
Brian Lee

**Comment by bendichter** [comment]
@ucscbrianlee sorry for the delay.

NWB is really built for time series data, as in you are sampling something at a fairly fast rate (<1 Hz) and storing that data over time. 

NWB and DANDI are really meant for neurophysiology data. You can store behavioral data too, but it really makes the most sense to use NWB and DANDI if you are storing that alongside measurements of neural activity.

If all you have is more like summary statistics of behavior, Zenodo may be a better option.

Did I understand your situation correctly? I'd be happy to discuss further if there is something I missed.

**Comment by ucscbrianlee** [comment]
@bendichter  

Thank you so much for your help.

Yes, agreed the summary data (.csv file) would be incompatible with the NWB format (thank you for your earlier help on that topic).

Part of this question is whether the ZebraLab .raw  file could not be in NWB.  There is some component in the .raw that is time series (i.e., each file has an embedded timecode as metadata). Although that is a proprietary file -there is a suggestion the lab might be able to convert it to a .avi movie file of the fish. The ZebraLab software outputs an Excel that has some time series info on the far right:

location | animal | user | sn | an | datatype | start | end | startreason | endreason | frect | fredur | midct | middur | burct | burdur | zerct | zerdur | actinteg | stdate | sttime
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
w001 | A01 | ZEB8139-4-PC\ZEB8139-4 | 1 | 0 | quantauc | 0 | 60 | End of period | End of period | 2 | 59.9 | 2 | 0.2 | 0 | 0 | 2 | 59.7 | 76 | 1/7/19 | 15:38:52
w002 | A02 | ZEB8139-4-PC\ZEB8139-4 | 1 | 0 | quantauc | 0 | 60 | End of period | End of period | 0 | 60 | 0 | 0 | 0 | 0 | 0 | 60 | 0 | 1/7/19 | 15:38:52  


I have feeling trying to put that into something like the below would not work:
```
    data=[1, -1, 1, -1],
    timestamps=[15.0, 30.0, 60.0, 95.0],
```

Mainly I want to confirm I am representing things correctly (to those desiring the files in DANDI). 

Thanks!
Brian

**Comment by ucscbrianlee** [comment]
Closing this as @DailyDreaming Lon Blauvelt and myself looked again at this data type and concluded that only through a NeuroConv script would this translate into NWB.  Since this data is not paired with ePhys/oPhys,  then as Ben Dichter the data does not really alight with the DANDI archive. 

Per discussion with Lon, a ticket at NeuroConv was created in case in the future there may be a need for NWB converstion of this data -not the case now.  https://github.com/catalystneuro/neuroconv/issues/1172
